web_monitoring/cli.py

```diff
@@ -1,51 +1,455 @@
+        session = ia.WaybackSession(**session_options)
+        self.wayback = ia.WaybackClient(session=session)
+
+    def is_active(self):
+        return not self.cancel.is_set()
+
+    def run(self):
+        """
+        Work through the queue of CDX records to load them from Wayback,
+        transform them to Web Monitoring DB import entries, and queue them for
+        importing.
+        """
+        while self.is_active():
+            try:
+                record = next(self.records)
+                self.summary['total'] += 1
+            except StopIteration:
+                break
+
+            self.handle_record(record, retry_connection_failures=True)
+
+        self.wayback.close()
+        return self.summary
+
+    def handle_record(self, record, retry_connection_failures=False):
+        """
+        Handle a single CDX record.
+        """
+        # Check for whether we already know this can't be played and bail out.
+        if self.unplaybackable is not None and record.raw_url in self.unplaybackable:
+            self.summary['playback'] += 1
+            return
+
+        try:
+            version = self.process_record(record, retry_connection_failures=True)
+            self.results_queue.put(version)
+            self.summary['success'] += 1
+        except ia.MementoPlaybackError as error:
+            self.summary['playback'] += 1
+            if self.unplaybackable is not None:
+                self.unplaybackable[record.raw_url] = datetime.utcnow()
+            logger.info(f'  {error}')
+        except requests.exceptions.HTTPError as error:
+            if error.response.status_code == 404:
+                logger.info(f'  Missing memento: {record.raw_url}')
+                self.summary['missing'] += 1
+            else:
+                # TODO: consider not logging this at a lower level, like debug
+                # unless failure_queue does not exist. Unsure how big a deal
+                # this error is to log if we are retrying.
+                logger.info(f'  (HTTPError) {error}')
+                if self.failure_queue:
+                    self.failure_queue.put(record)
+                else:
+                    self.summary['unknown'] += 1
+        except ia.WaybackRetryError as error:
+            logger.info(f'  {error}; URL: {record.raw_url}')
+
+            if self.failure_queue:
+                self.failure_queue.put(record)
+            else:
+                self.summary['unknown'] += 1
+        except Exception as error:
+            # FIXME: getting read timed out connection errors here...
+            # requests.exceptions.ConnectionError: HTTPConnectionPool(host='web.archive.org', port=80): Read timed out.
+            # TODO: don't count or log (well, maybe DEBUG log) if failure_queue
+            # is present and we are ultimately going to retry.
+            logger.exception(f'  ({type(error)}) {error}; URL: {record.raw_url}')
```

Comment:
A minor thing---using `repr` instead of `str` will give you the type and the args in one:
```suggestion
            logger.exception(f'  {error!r}; URL: {record.raw_url}')
```